code-challenges.test.js

```diff
@@ -35,8 +69,33 @@ var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+describe("a funciton called oddSort", () => {
+    it("takes in an array and returns an array with only odd numbers ordered least to greatest", () => {
+        expect(oddSort(fullArr1)).toEqual([-9, 7, 9, 199])
+        expect(oddSort(fullArr2)).toEqual([-823, 7, 23])
+    })
+})
 
 // b) Create the function that makes the test pass.
+//create a funciton that takes in an array of mixed elements
+//I need to iterate over the array
+//I need to filter out the odd numbers (+ & -)
+//add odd numbers to a new array
+//return the new array
+
+const oddSort = (array) => {
+    let oddArr = []
+     oddArr = array.filter(value => {
+        if (value % 2 !== 0 && typeof value !== "string" && typeof value !== "boolean")
+        return oddArr.push(value)
+    })
+        return oddArr.sort((a,b) => a-b);
+}
+
```

Comment:
Your logic looks good. The cool thing about using filter is that it returns an array. So you don't need to declare an array or push values. The other thing about filter is it has a built in if statement so you can just have your evaluation be the return statement.